### PR TITLE
First stab at Object type

### PR DESCRIPTION
**Pull request by LeaVerou**
Haven't tried it, feel free to merge if it works!

### CODE REVIEW

**Comment by DmitrySharabin** [comment]
Is there a need to trim values?

For example, `parse("foo: ba\\,r, key: value")` returns `{foo: " ba\\,r", "key": " value"}` where both values have spaces before the actual value.

**Comment by DmitrySharabin** [comment]
> Is there a need to trim values?

If we don't, `parse("key: false")` becomes `{ key: " false" }` instead of `{ key: false }`.

File: src/props/types.js [hunk 1]
```diff
@@ -131,4 +131,70 @@ types.set(Array, {
...
+types.set(Object, {
+	equals (a, b, { valueType } = {}) {
+		let aKeys = Object.keys(a);
+		let bKeys = Object.keys(b);
+
+		if (aKeys.length !== bKeys.length) {
+			return false;
+		}
+
+		return aKeys.every(key => equals(a[key], b[key], valueType));
```
Comment (DmitrySharabin) [inline src/props/types.js:145]: Is it a recursive call? Or do we expect to use some other `equals` function inside `every()`?

Comment (DmitrySharabin) [reply src/props/types.js:145]: I think we do, but it seems shadowed by the `equals` function we are in.

File: src/props/types.js [hunk 2]
```diff
@@ -131,4 +131,70 @@ types.set(Array, {
...
+	 * @param {boolean} [options.useMap] Whether to return a Map instead of an object
+	 * @param {string} [options.separator=","] The separator between entries.
+	 */
+	parse (value, { valueType, useMap, separator = ", " } = {}) {
+		separator = separator.trim();
+		let entrySeparatorRegex = RegExp(`\\s*(?<!\\\\)${separator}\\s*`);
+		let entries = value.trim().split(entrySeparatorRegex);
+
+		entries = entries.map(entry => {
+			let parts = entry.split(/(?<!\\):/);
```
Comment (DmitrySharabin) [inline src/props/types.js:164]:
```suggestion
			let parts = entry.split(/(?<!\\):\s*/);
```

This should fix issues with extra spaces.

**Comment by LeaVerou** [comment]
Gah, I removed the `\s*` from the regex because then re-joining is lossy, and was planning to trim manually (by calling `trim()`) but I forgot. Can you add that?

File: src/props/types.js [hunk 3]
```diff
@@ -131,4 +131,70 @@ types.set(Array, {
...
+			if (parts.length > 2) {
+				// Value contains colons
+				key = parts.shift();
+				value = parts.join(":");
+			}
+			else if (parts.length > 0) {
+				key = parts[0];
+				value = parts.length > 1 ? parts[1] : true;
+			}
+
```
Comment (DmitrySharabin) [inline src/props/types.js:176]:
```suggestion

			value = value.trim();
```

Comment (DmitrySharabin) [reply src/props/types.js:176]: Like so?

Comment (LeaVerou) [reply src/props/types.js:176]: That will fail if they're not strings. Just pushed a fix, lemme know if it works.

Comment (DmitrySharabin) [reply src/props/types.js:176]: I does work!

**Comment by LeaVerou** [comment]
Does that mean it can be merged?

**Comment by DmitrySharabin** [comment]
I tested it a bit. And yes, I think it can be merged. It took me some time to figure out how it works, though. :)

**Pull request merged**